### PR TITLE
[vscode][7/n] Setup Env Variables: Connect missing API keys error to command

### PR DESCRIPTION
**Pull request by rossdanlm**
[vscode][7/n] Setup Env Variables: Connect missing API keys error to command

Now it's more intuitive for users to know what needs to be done when they receive this error. I also moved the logic for the command into a utils file so it can be accessed both by activation flow and the error checking flow in the editor

## Test Plan

Getting missing API key error

https://github.com/lastmile-ai/aiconfig/assets/151060367/cc6c51e5-d205-4afa-a3b4-dc04f68d4946

Walkthrough commands

https://github.com/lastmile-ai/aiconfig/assets/151060367/bda40b3c-d260-40a2-82da-563d363152b3

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/lastmile-ai/aiconfig/pull/1300).
* __->__ #1300
* #1291

### CODE REVIEW

File: vscode-extension/src/aiConfigEditor.ts [hunk 1]
```diff
@@ -340,12 +341,23 @@ export class AIConfigEditorProvider implements vscode.CustomTextEditorProvider {
...
-            notification.title,
-            message ? "Details" : undefined
-          );
+          let notificationAction;
+          // TODO: Create a constant value somewhere in lastmile-utils to
+          // centralize string error message for missing API key. This
+          // logic is defined in https://github.com/lastmile-ai/aiconfig/blob/33fb852854d0bd64b8ddb4e52320112782008b99/python/src/aiconfig/util/config_utils.py#L41
+          if (message.includes("Missing API key")) {
+            notificationAction = await notificationFn(
+              "Looks like you're missing an API key, please set it in your env variables",
```
Comment (rholinshead) [inline vscode-extension/src/aiConfigEditor.ts:350]: One thing I found out recently is that vscode has MarkdownString which can be used in some vscode components. If it works, would be great to use to show documentation link here as well:

```
  const message = new vscode.MarkdownString();
  message.appendMarkdown(
    "Looks like you're missing an API key. Please set it in your <a href='https://aiconfig.lastmileai.dev/docs/getting-started/#setup-your-api-keys' target='_blank'>.env variables</a>"
  );
  message.isTrusted = true; // allow links to be clickable -- not sure if needed
```

**Review by rholinshead — APPROVED** [review]
LGTM, just suggestion to use markdown for link to docs if it'll work

File: vscode-extension/src/aiConfigEditor.ts (hunk 1 above)
Comment (rossdanlm) [reply vscode-extension/src/aiConfigEditor.ts:350]: I tried getting this to work but wasn't able to for diagnostics messaging. Seems like it's still an open issue: https://github.com/microsoft/vscode/issues/54272

**Comment by rossdanlm** [comment]
Updated content a bit

Screenshot 2024-02-22 at 17 45 53

**Pull request merged**